scania-digital-design-system/tegel
# feat(date-picker): inital date-picker development

Date Picker component.

**To-do**
- [ ] Review and align design
- [ ] UX questions: What happens when I click a date in the next month, should the dates be update to reflect the selected datemonth?
- [ ] Locales - date fns have support for locales, we need to see which are available and expose the available options as a prop on the component.
- [ ] Can the start/selected date be null?
- [ ] Make sure all available props in Date Picker is available in Date Range Picker
- [ ] Enable slots? 
- [ ] What data should be submitted in a form? 
- [ ] The focus state of the next/previous buttons?
- [ ] When should the Date Picker close? Only on click outside? What if you click the Date Picker and then click the input field.
- [ ] The Date Picker needs further control of the popover, it should open when you click the input field. But it should not close when the input field is clicked, if the Date Picker is opened. I do not think this is possible with the implementation of the popover we have today. 
- [x] Error state - prop to set the error state to true and slot to allow for users to pass in a error message.
- [x] Success state - prop to set the error state to true and slot to allow for users to pass in a success message.
- [x] Label & label position props
- [x] Min/Max props
- [x] Disable same date selection
- [x] Something buggy with choosing a date a couple of months ahead


**Solving issue**  _
Fixes: -

**How to test**  
1. Go to Date Picker
2. Check it out
3. Go to Date Range Picker
4. Check it out
